quartz_solar_forecast/data.py

```diff
@@ -37,16 +38,39 @@ def get_gfs_nwp(site: PVSite, ts:datetime) -> xr.Dataset:
     start = ts.date()
     end = start + pd.Timedelta(days=7)
 
-    # Getting GFS, from OPEN METEO
+    # Getting NWP, from OPEN METEO
+    urlStart = None
+    if source == "icon":
+        urlStart = "dwd-icon"
+    if source == "gfs":
+        urlStart = "gfs"
+
     url = (
```

Comment:
This is optional, but you could tidy this up and only pull the GFS visilibilty data when getting ICON data. 
This would mean not getting the rest of the GFS data when its not needed for ICON. 
Its also means we could easily add another nwp source if we wanted to. 

Could also leave this for now, as it works, and put that in a seperate issue/PR


```
if source == "icon"
    # get gfs data ....

if source == "gfs"
    # get icon data
    ....

    # pull gfs visibility data only
    .....
```